code/ui/Effects.cs

```diff
@@ -8,45 +8,40 @@
 
 namespace TTTReborn.UI
 {
+    using System.Linq;
+
     public class Effects : Panel
     {
-        private List<Effect> EffectList = new();
+        private readonly List<Effect> _effectList = new();
 
         public Effects()
         {
             StyleSheet.Load("/ui/Effects.scss");
-        }
-
-        public void AddEffect(TTTPerk perk)
-        {
-            Effect effect = new Effect(this);
-            effect.item = perk;
 
-            EffectList.Add(effect);
-        }
-
-        public void RemoveEffect(TTTPerk perk)
-        {
-            foreach (Effect effect in EffectList)
+            if (Local.Pawn is not TTTPlayer player)
             {
-                if (effect.item.Name == perk.Name)
-                {
-                    EffectList.Remove(effect);
-
-                    effect.Delete();
+                return;
+            }
 
-                    return;
-                }
+            PerksInventory perks = (player.Inventory as Inventory).Perks;
+            for (int i = 0; i < perks.Count(); i++)
+            {
+                AddEffect(perks.Get(i));
             }
         }
 
-        public void OnHotReloaded()
+        public void AddEffect(TTTPerk perk)
         {
-            PerksInventory perks = ((Local.Pawn as TTTPlayer).Inventory as Inventory).Perks;
+            _effectList.Add(new Effect(this) {item = perk});
+        }
 
-            for (int i = 0; i < perks.Count(); i++)
+        public void RemoveEffect(TTTPerk perk)
+        {
+            foreach (Effect effect in _effectList.Where(effect => effect.item.Name == perk.Name))
```

Comment:
Just one item of the same name can exist in a list, so we don't need a foreach loop theoretically. Maybe merge everything using Linq? :)